core/runtime.rs

```diff
@@ -647,6 +647,48 @@ impl JsRuntime {
     scope.perform_microtask_checkpoint();
   }
 
+  /// Waits for the given value to resolve while polling the event loop.
+  ///
+  /// This future resolves when either the value is resolved or the event loop runs to
+  /// completion.
+  pub async fn resolve_value(
+    &mut self,
+    global: v8::Global<v8::Value>,
+  ) -> Result<v8::Global<v8::Value>, AnyError> {
```

Comment:
@caspervonb can you please rework this function to `await_v8_promise`? It should assert that passed value is a `v8::Promise`. This is a very specialised API and I don't see why we should add path for values other than promises.

Reply:
So I started with a `resolve_promise` function that took a v8::Promise and failed if it wasn't a promise but that means you need to explicitly import v8 and convert in calling code, this lets you pass directly from execute_script with inferred types and matches the semantics of `Promise.resolve` (e.g `await Promise.resolve(0)` is basically equivalent to `resolve_promise(execute_script("(0)").unwrap()).unwrap().await`).

The main thing is taking in a `v8:::Global<v8::Value>>` and not a `v8:::Global<v8::Promise>>` as this means the caller doesn't have to import v8 and do manual conversions from the result returned by execute_script.

Reply:
Okay that makes sense, thanks